js_modules/dagit/packages/ui/src/components/__stories__/Toaster.stories.tsx

```diff
@@ -1,27 +1,40 @@
 // eslint-disable-next-line no-restricted-imports
-import {Intent, Position} from '@blueprintjs/core';
+import {Intent} from '@blueprintjs/core';
 import {Meta} from '@storybook/react';
 import * as React from 'react';
 
 import {Button} from '../Button';
 import {Group} from '../Group';
-import {GlobalToasterStyle, Toaster} from '../Toaster';
-
-const SharedToaster = Toaster.create({position: Position.TOP}, document.body);
+import {DToaster, DToasterShowProps, GlobalToasterStyle, Toaster} from '../Toaster';
 
 // eslint-disable-next-line import/no-default-export
 export default {
   title: 'Toaster',
 } as Meta;
 
 export const Sizes = () => {
+  const sharedToaster = React.useRef<DToaster | null>(null);
+
+  React.useEffect(() => {
+    const makeToaster = async () => {
+      sharedToaster.current = await Toaster.asyncCreate({position: 'top'}, document.body);
+    };
+    makeToaster();
+  }, []);
```

Comment:
Why not use your memoized toaster pattern?  

Another interesting pattern:
```suggestion
const sharedToasterRef = React.useMemo(() => {
  const ref = {current: null};
  (async () => {
    ref.current = await Toaster.asyncCreate({position: 'top'}, document.body);
  })();
  return ref;
}, []);

```